packages/kotti-ui/source/kotti-filter/types.ts

```diff
@@ -0,0 +1,155 @@
+import { Kotti } from '../types'
+import { SpecifyRequiredProps } from '../types/utilities'
+
+export namespace KottiFilter {
+	export enum ButtonLinkType {
+		DANGER = 'danger',
+		PRIMARY = 'primary',
+	}
+
+	export enum FilterType {
+		BOOLEAN = 'BOOLEAN',
+		DATE_RANGE = 'DATE_RANGE',
+		MULTI_ENUM = 'MULTI_ENUM',
+		NUMBER = 'NUMBER',
+		SEARCH = 'SEARCH',
+		SINGLE_ENUM = 'SINGLE_ENUM',
+		STRING = 'STRING',
+	}
+
+	export namespace Operation {
+		export enum Boolean {
+			EQUAL = 'EQUAL',
+		}
+
+		export enum Common {
+			IS_EMPTY = 'IS_EMPTY',
+		}
+
+		export enum DateRange {
+			IN_RANGE = 'IN_RANGE',
+		}
+
+		export enum MultiEnum {
+			ONE_OF = 'ONE_OF', // OR
+		}
+
+		export enum Number {
+			EQUAL = 'EQUAL',
+			GREATER_THAN = 'GREATER_THAN',
+			GREATER_THAN_OR_EQUAL = 'GREATER_THAN_OR_EQUAL',
+			LESS_THAN = 'LESS_THAN',
+			LESS_THAN_OR_EQUAL = 'LESS_THAN_OR_EQUAL',
+		}
+
+		export enum SingleEnum {
+			EQUAL = 'EQUAL',
+		}
+
+		export enum String {
+			CONTAINS = 'CONTAINS',
+		}
+	}
+
+	export type Column = {
+		key: string
+		label: string
+	} & (
+		| {
+				operations: (Operation.Boolean | Operation.Common)[]
+				type: FilterType.BOOLEAN
+		  }
+		| {
+				operations: (Operation.DateRange | Operation.Common)[]
+				type: FilterType.DATE_RANGE
+		  }
+		| {
+				operations: (Operation.MultiEnum | Operation.Common)[]
+				options: Kotti.FieldMultiSelect.Props['options']
+				type: FilterType.MULTI_ENUM
+		  }
+		| {
+				operations: (Operation.Number | Operation.Common)[]
+				type: FilterType.NUMBER
+		  }
+		| {
+				operations: Operation.String[]
```

Comment:
I think this is the first example for a case where `Operation.Common` wasn’t handled

Reply:
Actually, should `SEARCH` even have `operations`?